source/Cosmos.System.Plugs/System/IO/DirectoryInfoImpl.cs

```diff
@@ -36,13 +36,19 @@ public static class DirectoryInfoImpl
             Global.mFileSystemDebugger.SendInternal(aPath);
 
             aStorage = VFSManager.GetDirectory(aPath);
+
+            if (aStorage == null)

```

Comment:
But it throws a NullReferenceException if it doesn't, because the DirectoryEntry is null and it can't get the path